File: hoopBack/src/main/java/com/hoop/hoopback/entity/Notification.java
package com.hoop.hoopback.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recipient_id", nullable = false)
    private User recipient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "actor_id")
    private User actor; // The user who performed the action

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationType type;

    private Long entityId; // postId / gameId / etc.
    private String entityType; // "POST" / "GAME" / "USER"

    @Builder.Default
    private boolean isRead = false;

    @CreationTimestamp
    private LocalDateTime createdAt;

    public void markRead() {
        this.isRead = true;
    }
}

File: hoopFront/src/types/notification.ts
import type { UserSummary } from './user';

export type NotificationType = 'LIKE' | 'COMMENT' | 'FOLLOW' | 'REPOST' | 'GAME_JOIN';

export interface Notification {
    id: number;
    type: NotificationType;
    actor: UserSummary | null;
    entityId: number | null;
    entityType: string | null;
    isRead: boolean;
    createdAt: string;
}
